From a53069b6bf20021c2eb973688a711997053a3831 Mon Sep 17 00:00:00 2001
From: Billy Tsai <billy_tsai@aspeedtech.com>
Date: Tue, 21 Feb 2023 18:07:46 +0800
Subject: [PATCH 4/4] soc: ast10x0: Clear the higher priority pinctrl setting.

This commit clears the higher priority setting for hvi3c, which will
avoid the pinctrl setting error under non AC power on conditions.

Signed-off-by: Billy Tsai <billy_tsai@aspeedtech.com>
Change-Id: I9ed4261a331edd358e4848d3410f59fc7cee9581
---
 soc/arm/aspeed/ast10x0/sig_def_list.h | 24 ++++++++++++++++--------
 1 file changed, 16 insertions(+), 8 deletions(-)

diff --git a/soc/arm/aspeed/ast10x0/sig_def_list.h b/soc/arm/aspeed/ast10x0/sig_def_list.h
index 954ea8906c..c0d2ff6bef 100644
--- a/soc/arm/aspeed/ast10x0/sig_def_list.h
+++ b/soc/arm/aspeed/ast10x0/sig_def_list.h
@@ -85,29 +85,37 @@ SIG_DEFINE(MTDO2,   B12,    SIG_DESC_SET(0x41C, 24))
 #if DT_NODE_HAS_STATUS(DT_NODELABEL(i3c0), okay)
 SIG_DEFINE(I3C1SCL, J16, SIG_DESC_SET(0x418, 16), SIG_DESC_CLEAR(0x4B8, 8))
 SIG_DEFINE(I3C1SDA, J15, SIG_DESC_SET(0x418, 17), SIG_DESC_CLEAR(0x4B8, 9))
-SIG_DEFINE(HVI3C1SCL, M15, SIG_DESC_SET(0x4B8, 8), SIG_DESC_CLEAR(0x418, 16))
-SIG_DEFINE(HVI3C1SDA, M16, SIG_DESC_SET(0x4B8, 9), SIG_DESC_CLEAR(0x418, 17))
+SIG_DEFINE(HVI3C1SCL, M15, SIG_DESC_SET(0x4B8, 8), SIG_DESC_CLEAR(0x418, 16),
+	   SIG_DESC_CLEAR(0x418, 8))
+SIG_DEFINE(HVI3C1SDA, M16, SIG_DESC_SET(0x4B8, 9), SIG_DESC_CLEAR(0x418, 17),
+	   SIG_DESC_CLEAR(0x418, 9))
 #endif
 
 #if DT_NODE_HAS_STATUS(DT_NODELABEL(i3c1), okay)
 SIG_DEFINE(I3C2SCL, J14, SIG_DESC_SET(0x418, 18), SIG_DESC_CLEAR(0x4B8, 10))
 SIG_DEFINE(I3C2SDA, H16, SIG_DESC_SET(0x418, 19), SIG_DESC_CLEAR(0x4B8, 11))
-SIG_DEFINE(HVI3C2SCL, L14, SIG_DESC_SET(0x4B8, 10), SIG_DESC_CLEAR(0x418, 18))
-SIG_DEFINE(HVI3C2SDA, K15, SIG_DESC_SET(0x4B8, 11), SIG_DESC_CLEAR(0x418, 19))
+SIG_DEFINE(HVI3C2SCL, L14, SIG_DESC_SET(0x4B8, 10), SIG_DESC_CLEAR(0x418, 18),
+	   SIG_DESC_CLEAR(0x418, 10))
+SIG_DEFINE(HVI3C2SDA, K15, SIG_DESC_SET(0x4B8, 11), SIG_DESC_CLEAR(0x418, 19),
+	   SIG_DESC_CLEAR(0x418, 11))
 #endif
 
 #if DT_NODE_HAS_STATUS(DT_NODELABEL(i3c2), okay)
 SIG_DEFINE(I3C3SCL, H15, SIG_DESC_SET(0x418, 20), SIG_DESC_CLEAR(0x4B8, 12))
 SIG_DEFINE(I3C3SDA, G15, SIG_DESC_SET(0x418, 21), SIG_DESC_CLEAR(0x4B8, 13))
-SIG_DEFINE(HVI3C3SCL, L16, SIG_DESC_SET(0x4B8, 12),  SIG_DESC_CLEAR(0x418, 20))
-SIG_DEFINE(HVI3C3SDA, K16, SIG_DESC_SET(0x4B8, 13),  SIG_DESC_CLEAR(0x418, 21))
+SIG_DEFINE(HVI3C3SCL, L16, SIG_DESC_SET(0x4B8, 12),  SIG_DESC_CLEAR(0x418, 20),
+	   SIG_DESC_CLEAR(0x418, 12))
+SIG_DEFINE(HVI3C3SDA, K16, SIG_DESC_SET(0x4B8, 13),  SIG_DESC_CLEAR(0x418, 21),
+	   SIG_DESC_CLEAR(0x418, 13))
 #endif
 
 #if DT_NODE_HAS_STATUS(DT_NODELABEL(i3c3), okay)
 SIG_DEFINE(I3C4SCL, J13, SIG_DESC_SET(0x418, 22), SIG_DESC_CLEAR(0x4B8, 14))
 SIG_DEFINE(I3C4SDA, G16, SIG_DESC_SET(0x418, 23), SIG_DESC_CLEAR(0x4B8, 15))
-SIG_DEFINE(HVI3C4SCL, K14, SIG_DESC_SET(0x4B8, 14),  SIG_DESC_CLEAR(0x418, 22))
-SIG_DEFINE(HVI3C4SDA, K15, SIG_DESC_SET(0x4B8, 15),  SIG_DESC_CLEAR(0x418, 23))
+SIG_DEFINE(HVI3C4SCL, K14, SIG_DESC_SET(0x4B8, 14),  SIG_DESC_CLEAR(0x418, 22),
+	   SIG_DESC_CLEAR(0x418, 14))
+SIG_DEFINE(HVI3C4SDA, K15, SIG_DESC_SET(0x4B8, 15),  SIG_DESC_CLEAR(0x418, 23),
+	   SIG_DESC_CLEAR(0x418, 15))
 #endif
 
 #if DT_NODE_HAS_STATUS(DT_NODELABEL(i2cfilter), okay) && CONFIG_I2C_PFR_FILTER
-- 
2.25.1

